Set timestamp on bufferedport - Yarp bindings in C# - Error

Hi all,

I tried to set a time stamp in a bufferedport. I'm using the yarp bindings for C#.

    Stamp ts = new Stamp(); 
    outputport.setEnvelope(ts);

I get the folowwing error:

    error CS1503: Argument #1 cannot convert `Stamp' expression to type `PortWriter'

@stephane-lallee @paulfitz, any idea?

--- a/bindings/tests/csharp/TestStamp.cs
+++ b/bindings/tests/csharp/TestStamp.cs
@@ -1,0 +1,25 @@
+using System;
+
+// Copyright: (C) 2015 iCub Facility
+// Author: Paul Fitzpatrick
+// CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ 
+// I don't really know C#, just bluffing from Java
+
+namespace HelloNameSpace
+{
+    public class TestStamp
+    {
+	static void Main(string[] args)
+	{
+	    Network.init();
+	    BufferedPortBottle p = new BufferedPortBottle();
+	    if (!p.open("/csharp")) System.Environment.Exit(1);
+	    Stamp ts = new Stamp(); 
+	    p.setEnvelope(ts);
+	    p.close();
+	    Network.fini();
+	}
+    }
+}
+

--- a/bindings/tests/csharp/run.sh
+++ b/bindings/tests/csharp/run.sh
@@ -16,6 +16,8 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 echo "HELLO $CSLIB // $DIR"
 
 echo "We compile using gmcs, the Mono C# Compiler"
-rm -rf test_string.exe
-gmcs -out:test_string.exe $DIR/TestString.cs $CSLIB/*.cs
-./test_string.exe
+for f in TestString TestStamp; do
+    rm -rf $f.exe
+    gmcs -out:$f.exe $DIR/$f.cs $CSLIB/*.cs
+    ./$f.exe
+done

--- a/bindings/yarp.i
+++ b/bindings/yarp.i
@@ -641,6 +641,12 @@ typedef yarp::os::BufferedPort<ImageRgbFloat> BufferedPortImageRgbFloat;
 	}
 }
 
+%extend yarp::os::Contactable {
+  bool setEnvelope(Portable& data) {
+    return self->setEnvelope(*((PortWriter*)(&data)));
+  }
+}
+
 //////////////////////////////////////////////////////////////////////////
 // Deal with PolyDriver idiom that doesn't translate too well
 

